aten/src/ATen/detail/XPUHooksInterface.h

```diff
@@ -43,10 +43,6 @@ struct TORCH_API XPUHooksInterface {
     TORCH_CHECK(false, "Cannot get XPU global device index without ATen_xpu library.");
   }
 
-  virtual Generator getXPUGenerator(C10_UNUSED DeviceIndex device_index = -1) const {
```

Comment:
Why is this removed?

Reply:
Change back. Using hooks instead of ifdef.